### PR TITLE
[DX-155] Handle undefined/null query params in get requests

### PR DESCRIPTION
**Pull request by nickbalestra**
We probably want to have a proper sanitizer for the version when used within the url of the components endpoints.

### CODE REVIEW

**Comment by nickbalestra** [comment]
Atm doing a get on `/component/^1.2.3/` (escaped ofc) brake the endpoint

**Comment by nickbalestra** [comment]
@matteofigus added some utils, and test bot for the utils and for the whole `href-builder` module

File: test/unit/client-href-builder.js [hunk 1]
```diff
@@ -87,6 +87,26 @@ describe('client : href-builder :', () => {
       });
     });
 
+    describe('when some of the paramaters are undefined or null', () => {
```
Comment (matteofigus) [inline test/unit/client-href-builder.js]: Actually, `null` is different from `undefined`. `null` is valid in JSON and we should keep it blank if it's there. I believe the only thing we should tackle here is `undefined`.

**Comment by nickbalestra** [comment]
Yes, and also address NaN

File: test/unit/client-href-builder.js (hunk 1 above)
Comment (nickbalestra) [reply test/unit/client-href-builder.js]: NaN

**Pull request merged**